src/services/inlayHints.ts

```diff
@@ -750,6 +751,17 @@ export function provideInlayHints(context: InlayHintsContext): InlayHint[] {
                         visitForDisplayParts(node.type);
                     }
                     break;
+                case SyntaxKind.IndexSignature:
+                    Debug.assertNode(node, isIndexSignatureDeclaration);
+                    Debug.assertEqual(node.parameters.length, 1);

```

Comment:
I'm not sure if an assertion is appropriate - what happens if you have something like

```ts
const x: { []: string }
const y = x;
```

Reply:
Specifically, assertions shouldn't assume well-formed code, they should assert invariants of how the compiler constructs the tree. You'd probably be better off with just

```
node.parameters.forEach(visitForDisplayParts)
```

Reply:
I get `An index signature must have exactly one parameter.ts(1096)` for that code. Also, the parser doesn't create an index signature node for that (the type literal node just has no `member` children nodes) so this switch case isn't hit,

That being said, I also wasn't so enthusiastic about using the assert but wasn't sure about alternatives. Your suggestion is better though since this is a user error and we already report diagnostics to the user elsewhere.

I think this should be represented in the types though so we can enforce the invariant. Can we just add something like:
```ts
export interface IndexSignatureDeclaration extends SignatureDeclarationBase, ...
{
    ...
    readonly parameters: [ParameterDeclaration] & NodeArray<ParameterDeclaration>;
}
```

It's probably not that easy since this is a public API and would be breaking.